qml/platform.ubports/FileSelectorPL.qml

```diff
@@ -18,17 +18,69 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-import QtQuick 2.0
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import Ubuntu.Content 1.3
+import "."
 
-Item {
+Page {
     id: fs
-
+    
     property string selectedFilepath
     property var nameFilters
     signal selected
+    signal cancel()    
+    
+	property var activeTransfer
+	property var url
+	property var handler
+	property var contentType
+
+    header: HeaderBarImpl {
+        title: i18n.tr("Choose")
+        page: fs
+    }
+
+    ContentPeerPicker {
+        anchors.fill: parent
+        anchors.topMargin: fs.header.height
+        visible: parent.visible
+        showTitle: false
+        contentType: ContentType.Documents
+        handler: ContentHandler.Source
+
+        onPeerSelected: {
+            peer.selectionType = ContentTransfer.Single
+            fs.activeTransfer = peer.request()
+            fs.activeTransfer.stateChanged.connect(function() {
+				if (fs.activeTransfer.state === ContentTransfer.InProgress) {
+					console.log("In progress");
+					fs.activeTransfer.items = fs.activeTransfer.items[0].url = url;
+					fs.activeTransfer.state = ContentTransfer.Charged;
+				}
+                if (fs.activeTransfer.state === ContentTransfer.Charged) {
+					console.log("Charged");
+                    selectedFilepath = fs.activeTransfer.items[0].url;
+                    selectedFilepath = selectedFilepath.replace("file://", "");
+                    fs.selected();
+                    app.pages.pop(Qt.resolvedUrl("../qml/platform/FileSelectorPL.qml"));
+                }
+            })
+        }
+
+        onCancelPressed: {
+            pageStack.pop()
```

Comment:
we don't call `pageStack` directly throughout the code. Let's keep it and use `app.pages.pop()` instead.